File: core/src/main/java/net/techcable/sonarpet/nms/entity/generators/EntityUndeadPetGenerator.java
package net.techcable.sonarpet.nms.entity.generators;

import net.techcable.sonarpet.utils.Versioning;

import org.objectweb.asm.Type;

import static org.objectweb.asm.Opcodes.*;

public class EntityUndeadPetGenerator extends EntityPetGenerator {
    public EntityUndeadPetGenerator(Type currentType, Class<?> hookClass, Class<?> entityClass) {
        super(currentType, hookClass, entityClass);
    }

    @Override
    protected void generate0() {
        super.generate0();
        final String livingUpdateMethod = Versioning.NMS_VERSION.getObfuscatedMethod("LIVING_UPDATE_METHOD");
        // Prevent the pet from burning
        generator.generateMethod(
                (generator) -> {},
                ACC_PUBLIC,
                livingUpdateMethod,
                Type.VOID_TYPE
        );
    }
}

File: core/src/main/kotlin/net/techcable/sonarpet/nms/entity/generators/EntityUndeadPetGenerator.kt
package net.techcable.sonarpet.nms.entity.generators

import com.dsh105.echopet.compat.api.plugin.IEchoPetPlugin
import com.google.common.collect.ImmutableList
import org.objectweb.asm.Type
import net.techcable.sonarpet.utils.*
import org.objectweb.asm.Type.*

class EntityUndeadPetGenerator(plugin: IEchoPetPlugin, currentType: Type, hookClass: Class<*>, entityClass: Class<*>) : EntityPetGenerator(plugin, currentType, hookClass, entityClass) {
    override val generatedMethods: ImmutableList<GeneratedMethod>
        // Disable setOnFire to prevent the pet from burning
        get() = super.generatedMethods + GeneratedMethod.noOp("setOnFire", parameterTypes = listOf(INT_TYPE))
}